go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -424,10 +425,55 @@ func (se *Engine) reload(ctx context.Context, includeStats bool) error {
 		return err
 	}
 
+	innodbTablesStats := make(map[string]*Table)
+	if includeStats {
+		if innodbTableSizesQuery := conn.Conn.BaseShowInnodbTableSizes(); innodbTableSizesQuery != "" {
+			// Since the InnoDB table size query is available to us on this MySQL version, we should use it.
+			// We therefore don't want to query for table sizes in getTableData()
+			includeStats = false
+
+			innodbResults, err := conn.Conn.Exec(ctx, innodbTableSizesQuery, maxTableCount, false)
+			if err != nil {
+				return vterrors.Wrapf(err, "in Engine.reload(), reading innodb tables")
+			}
+			for _, row := range innodbResults.Rows {
+				innodbTableName := row[0].ToString() // In the form of encoded `schema/table`
+				fileSize, _ := row[1].ToCastUint64()
+				allocatedSize, _ := row[2].ToCastUint64()
+
+				if _, ok := innodbTablesStats[innodbTableName]; !ok {
+					innodbTablesStats[innodbTableName] = &Table{}
+				}
+				// There could be multiple appearances of the same table in the result set:
+				// A table that has FULLTEXT indexes will appear once for the table itself,
+				// with total size of row data, and once for the aggregates size of all
+				// FULLTEXT indexes. We aggregate the sizes of all appearances of the same table.
+				table := innodbTablesStats[innodbTableName]
+				table.FileSize += fileSize
+				table.AllocatedSize += allocatedSize
+
+				if strings.Contains(innodbTableName, "#p#") {
+					// innodbTableName is encoded any special characters are turned into some @0-f0-f0-f value.
```

Comment:
Nit, but this could be improved to:
```
				// innodbTableName is encoded any special characters are turned into some @0-f0-f0-f value.
				// Therefore this "#p#" here is a clear indication that we are looking at a partitioned table.
				if originalTableName, _, found := strings.Cut(innodbTableName, "#p#"); found {
					// We turn `my@002ddb/tbl_part#p#p0` into `my@002ddb/tbl_part`
					// and aggregate the total partition sizes.
					if _, ok := innodbTablesStats[originalTableName]; !ok {
						innodbTablesStats[originalTableName] = &Table{}
						originalTable := innodbTablesStats[originalTableName]
						originalTable.FileSize += fileSize
						originalTable.AllocatedSize += allocatedSize
					}
				}
```

Reply:
The suggested rewrite is incorrect, unfortunately. The deal is that we will have, say, 8 partitions for the same table. Therefore we will have one time where `!ok` holds true, and then `7` more times where `ok == true`, in which case we should proceed to `+= fileSize` and `+= allocatedSize`. But with this rewrite we will skip those `7` times.

Reply:
Ah, got it. I misunderstood earlier. Making the change.

Reply:
Done.